data-conversion/src/main.rs

```diff
@@ -0,0 +1,171 @@
+            }
+        }
+    }
+
+    eprintln!(
+        "Successfully converted from {} to {} in {}",
+        convert_from, convert_to, filepath_send
+    );
+}
+
+/// Formats [to_format] properly
+fn format_binary(to_format: u32) -> String {
+    let binary_str = format!("{:032b}", to_format);
+    format!(
+        "{} {} {}",
+        &binary_str[0..1], // Sign bit
+        &binary_str[1..9], // Exponent
+        &binary_str[9..]   // Significand
+    )
+}
+
+fn format_hex(to_format: u32) -> String {
+    let formatted_hex_str = format!("{:X}", to_format);
+    format!("0x{}", &formatted_hex_str)
+}
+
+/// Converts [binary_string] to binary and appends to [filepath_send]
+fn float_to_binary(
+    binary_string: &str,
+    filepath_send: &mut File,
+) -> std::io::Result<()> {
+    let float_of_string: f32;
+    // Convert string to float
+    match binary_string.parse::<f32>() {
+        Ok(parsed_num) => float_of_string = parsed_num,
+        Err(_) => {
+            panic!("Failed to parse float from string")
+        }
+    }
+
+    // Convert float to binary
+    let binary_of_float = float_of_string.to_bits();
+    let formatted_binary_str = format_binary(binary_of_float);
+
+    // Write binary string to the file
+    filepath_send.write_all(formatted_binary_str.as_bytes())?;
+    filepath_send.write_all(b"\n")?;
+
+    Ok(())
+}
+
+/// Converts [hex_string] to binary and appends to [filepath_send]
+fn hex_to_binary(hex_string: &str, filepath_send: &mut File) -> io::Result<()> {
+    // Convert hex to binary
+    let binary_of_hex = match u32::from_str_radix(hex_string, 16) {
+        Ok(value) => value,
+        Err(err) => {
+            return Err(io::Error::new(io::ErrorKind::InvalidData, err))
+        }
+    };
+
+    // Format nicely
+    let formatted_binary_str = format!("{:b}", binary_of_hex);
+
+    // Write binary string to the file
+    filepath_send.write_all(formatted_binary_str.as_bytes())?;
+    filepath_send.write_all(b"\n")?;
+
+    Ok(())
+}
+
+fn binary_to_hex(
+    binary_string: &str,
+    filepath_send: &mut File,
+) -> io::Result<()> {
+    let hex_of_binary = match u32::from_str_radix(binary_string, 2) {
+        Ok(value) => value,
+        Err(err) => {
+            return Err(io::Error::new(io::ErrorKind::InvalidData, err))
+        }
+    };
+
+    let formatted_hex_str = format_hex(hex_of_binary);
+
+    filepath_send.write(formatted_hex_str.as_bytes())?;
+    filepath_send.write_all(b"\n")?;
+
+    Ok(())
+}
+
+fn binary_to_float(
+    binary_string: &str,
+    filepath_send: &mut File,
+) -> io::Result<()> {
+    let binary_value = match u32::from_str_radix(binary_string, 2) {
+        Ok(value) => value,
+        Err(err) => {
+            return Err(io::Error::new(io::ErrorKind::InvalidData, err))
+        }
+    };
+
+    // Interpret the integer as the binary representation of a floating-point number
+    let float_value = unsafe { std::mem::transmute::<u32, f32>(binary_value) };
```

Comment:
I think you can just use [`f32::from_bits`](https://doc.rust-lang.org/std/primitive.f32.html#method.from_bits) for this, which doesn't require `unsafe`.